blockchain/chain_sync/src/network_context.rs

```diff
@@ -39,10 +39,10 @@ pub(crate) struct SyncNetworkContext<DB> {
 
     /// Manages peers to send requests to and updates request stats for the respective peers.
     pub peer_manager: Arc<PeerManager>,
-    db: Arc<DB>,
+    db: DB,
```

Comment:
This should be boxed. Otherwise it would need to be unpinned.
```suggestion
    db: Box<DB>,
```

Reply:
`DB` is already an `Arc` therefore it implements `Unpin` already, why should we introduce another smart pointer?

Reply:
`DB` doesn't have to contain an `Arc`. It can be anything. And we don't want to put anything in a future.